### PR TITLE
Dont show changelog msg for primary origin if it wasn't updated

### LINKED ISSUE #5333: Suppress DS consistent hash query params change log entry if nothing changed
## I'm submitting a ...
-  improvement request (usability, performance, tech debt, etc.)

## Traffic Control components affected ...
-  Traffic Ops

## Current behavior:
Every time a DS is created or updated, the following change log entry is created when no consistent hash query params are created.

```
DS: my-ds-xml-id, ID: 999, ACTION: Created 0 consistent hash query params
```

Doesn't make much sense to me.

## New behavior:
Don't do that if nothing changed with regards to a delivery service's consistent hash query params. If it went from 0 to 1 or 1 to 0 or 2 to 4, then the change log makes sense, otherwise, nobody cares. :)

## Minimal reproduction of the problem with instructions:
Create or update a delivery service. Notice the consistent hash query params change log entry.

### LINKED ISSUE #5335: Suppress "Updated primary origin" changelog entry if DS orgServerFqdn is unchanged
## I'm submitting a ...
-  improvement request (usability, performance, tech debt, etc.)

## Traffic Control components affected ...
-  Traffic Ops

## Current behavior:
When updating a delivery service, a changelog entry is created that says "DS: foo, ID: 442, ACTION: Updated primary origin: foo" even if the DS's `orgServerFqdn` wasn't changed.

## New behavior:
The aforementioned changelog entry should only be created if the DS's `orgServerFqdn` is changed. 

## Minimal reproduction of the problem with instructions:
Create a delivery service then make any update it. See the changelog entries.

### PR DESCRIPTION
**Pull request by srijeet0406**
## What does this PR (Pull Request) do?

- [x] This PR fixes #5335 and fixes #5333  

## Which Traffic Control components are affected by this PR?

- Traffic Ops
- CI tests

## What is the best way to verify this PR?

Try updating an existing DS, by changing anything other than the org server fqdn -> you should just see one changelog entry in TP corresponding to the update of the DS.
Now try updating the org server fqdn -> you should see two changelog entries now, one for the DS, and one for the primary origin of the DS being updated.
Also, note that we no longer produce changelog entries for consistent hash query params changes.
## If this is a bug fix, what versions of Traffic Control are affected?

- master
## The following criteria are ALL met by this PR

- [x] This PR includes tests
- [x] This PR does not include documentation
- [x] This PR includes an update to CHANGELOG.md
- [x] This PR includes any and all required license headers
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

**Comment by ocket8888** [comment]
Restarting CiaB test

File: traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go [hunk 1]
```diff
@@ -49,6 +49,14 @@ type TODeliveryService struct {
 	tc.DeliveryServiceNullableV30
 }
 
+// TODeliveryServiceOldDetails is the struct to store the old details while updating a DS
```
Comment (ocket8888) [inline traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go:52]: GoDoc should end with punctuation.

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 ### Added
 - Traffic Portal: [#5394](https://github.com/apache/trafficcontrol/issues/5394) - Converts the tenant table to a tenant tree for usability
 - Traffic Portal: [#5360](https://github.com/apache/trafficcontrol/issues/5360) - Adds the ability to clone a topology
+- Traffic Portal: upgraded delivery service UI tables to use more powerful/performant ag-grid component
```
Comment (mitchell852) [inline CHANGELOG.md:10]: probably not part of your PR, right?

File: CHANGELOG.md [hunk 3]
```diff
@@ -16,6 +17,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Traffic Ops: Added validation to ensure that the `weight` parameter of `parent.config` is a float
 
 ### Fixed
+- [#5335](https://github.com/apache/trafficcontrol/issues/5335) - Don't create a changelog entry for the primary origin if it hasn't really changed
```
Comment (mitchell852) [inline CHANGELOG.md:20]: nit: can u change to:

```
Don't create a change log entry if the delivery service primary origin hasn't changed
Don't create a change log entry for any delivery service consistent hash query params updates
```

File: traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go [hunk 4]
```diff
@@ -844,22 +849,23 @@ func updateV31(w http.ResponseWriter, r *http.Request, inf *api.APIInfo, dsV31 *
...
 			return nil, errCode, userErr, sysErr
 		}
 		exists, err := getSSLVersion(*ds.XMLID, tx)
 		if err != nil {
 			return nil, http.StatusInternalServerError, nil, fmt.Errorf("querying delivery service with sslKeyVersion failed: %s", err)
 		}
-		if exists && (oldCDNName != *ds.CDNName || oldRoutingName != *ds.RoutingName) {
-			return nil, http.StatusBadRequest, errors.New("delivery service has ssl keys that cannot be automatically changed, therefore CDN and routing name are immutable"), nil
+		if ds.CDNName != nil && ds.RoutingName != nil {
+			if exists && (oldDetails.OldCdnName != *ds.CDNName || oldDetails.OldRoutingName != *ds.RoutingName) {
```
Comment (mitchell852) [inline traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go:866]: i know u didn't create this variable but think you can change `exists` to `sslKeysExist` to make it semantic?

**Pull request merged**